src/Hal/Report/Html/Reporter.php

```diff
@@ -105,10 +138,40 @@ public function generate(Metrics $metrics)
 
         // json data
         file_put_contents(
-            $logDir . '/js/classes.js',
+            $logDir . '/classes.js',
             'var classes = ' . json_encode($consolidated->getClasses(), JSON_PRETTY_PRINT)
         );
 
+        // consolidated by groups
+        foreach ($consolidatedGroups as $name => $consolidated) {
+            $outDir = $logDir . DIRECTORY_SEPARATOR . $name;
+            $this->currentGroup = $name;
+            $this->assetPath = '../';
+
+            if (!file_exists($outDir)) {
+                mkdir($outDir, 0755, true);
+            }
+
+            $this->renderPage($this->templateDir . '/html_report/index.php', $outDir . '/index.html', $consolidated, $history);
+            $this->renderPage($this->templateDir . '/html_report/loc.php', $outDir . '/loc.html', $consolidated, $history);
+            $this->renderPage($this->templateDir . '/html_report/relations.php', $outDir . '/relations.html', $consolidated, $history);
+            $this->renderPage($this->templateDir . '/html_report/coupling.php', $outDir . '/coupling.html', $consolidated, $history);
+            $this->renderPage($this->templateDir . '/html_report/all.php', $outDir . '/all.html', $consolidated, $history);
+            $this->renderPage($this->templateDir . '/html_report/oop.php', $outDir . '/oop.html', $consolidated, $history);
+            $this->renderPage($this->templateDir . '/html_report/complexity.php', $outDir . '/complexity.html', $consolidated, $history);
+            $this->renderPage($this->templateDir . '/html_report/panel.php', $outDir . '/panel.html', $consolidated, $history);
+            $this->renderPage($this->templateDir . '/html_report/violations.php', $outDir . '/violations.html', $consolidated, $history);
+            $this->renderPage($this->templateDir . '/html_report/packages.php', $outDir . '/packages.html', $consolidated, $history);
+            $this->renderPage($this->templateDir . '/html_report/package_relations.php', $outDir . '/package_relations.html', $consolidated, $history);
+            $this->renderPage($this->templateDir . '/html_report/composer.php', $outDir . '/composer.html', $consolidated, $history);
```

Comment:
Do you think it could be a little be less verbose if you can
- Export the `$this->templateDir . '/html_report'` string outside the `foreach loop` ;
- Loop over all page name as each line are identical ?